graalpython/com.oracle.graal.python/src/com/oracle/graal/python/builtins/modules/BuiltinFunctions.java

```diff
@@ -286,6 +297,211 @@ public Object absObject(VirtualFrame frame, Object object,
         }
     }
 
+    /**
+     * Common class for all() and any() operations, as their logic and behaviors are very similar.
+     */
+    abstract static class AllOrAnyNode extends PNodeWithContext {
+        enum NodeType {
+            ALL,
+            ANY
+        }
+
+        @Child private PyObjectIsTrueNode isTrueNode = PyObjectIsTrueNode.create();
+
+        final private LoopConditionProfile loopConditionProfile = LoopConditionProfile.create();
+
+        abstract boolean execute(Frame frame, Object storageObj, NodeType nodeType);
+
+        @Specialization
+        boolean doBoolSequence(VirtualFrame frame,
+                        BoolSequenceStorage sequenceStorage,
+                        NodeType nodeType) {
+            boolean[] internalArray = sequenceStorage.getInternalBoolArray();
+
+            for (int i = 0; loopConditionProfile.profile(i < sequenceStorage.length()); i++) {
```

Comment:
Thanks for adding the profiles. This is already higher standard than in most of our loops elsewhere in the code. For this specific case of counted loop, you can also use:
```
int len = sequenceStorage.length();
loopConditionProfile.profileCounted(len);
for (..., loopConditionProfile.inject(i < len), ...)
```
the `loopConditionProfile.profile` is incrementing a counter (look at the source), which are unnecessary instructions in the loop if we know the number of the iterations upfront. `inject` is intrinsic that only communicates the probability of branch taken/not taken to the compiler, but otherwise a no-op. One could argue that this is stretching it a bit too far, because the counter increment may be optimized away, but it is at least the best practice and a canonical example from the JavaDocs of `LoopConditionProfile`

Reply:
Sounds good, changed in 66661b4. Thanks for the explanation as well

Reply:
Looks good now, thanks